pkg/controller/podautoscaler/metrics/metrics_client_test.go

```diff
@@ -184,10 +184,12 @@ func buildPod(namespace, podName string, podLabels map[string]string, phase api.
 }
 
 func (tc *testCase) verifyResults(t *testing.T, val *float64, timestamp time.Time, err error) {
-	assert.Equal(t, tc.desiredError, err)
 	if tc.desiredError != nil {
+		assert.Error(t, err)
+		assert.Contains(t, fmt.Sprintf("%v", err), fmt.Sprintf("%v", tc.desiredError))
 		return
 	}
+	assert.NoError(t, err)
```

Comment:
Please add test case for hint (add case w/ one pod missing and set desired error to contain the missing pod name)